plugins/defaultplugins/srplugin/vppcalls/srv6.go

```diff
@@ -0,0 +1,510 @@
+		log.Debugf("%v steering for l3 traffic with destination %v to SR policy (binding SID %v, policy index %v)",
+			operationProgressing, steering.L3Traffic.PrefixAddress, steering.PolicyBSID, steering.PolicyIndex)
+	} else {
+		log.Debugf("%v steering for l2 traffic from interface %v to SR policy (binding SID %v, policy index %v)",
+			operationProgressing, steering.L2Traffic.InterfaceName, steering.PolicyBSID, steering.PolicyIndex)
+	}
+
+	// converting policy reference
+	bsidAddr := make([]byte, 0)
+	if len(strings.Trim(steering.PolicyBSID, " ")) > 0 {
+		bsid, err := parseIPv6(steering.PolicyBSID)
+		if err != nil {
+			return fmt.Errorf("can't parse binding SID \"%v\"to IP address: %v ", steering.PolicyBSID, err)
+		}
+		bsidAddr = []byte(bsid)
+	}
+
+	// converting target traffic info
+	var prefixAddr []byte
+	steerType := SteerTypeIPv6
+	tableID := uint32(0)
+	maskWidth := uint32(0)
+	intIndex := uint32(0)
+	if steering.L3Traffic != nil {
+		ip, ipnet, err := net.ParseCIDR(steering.L3Traffic.PrefixAddress)
+		if err != nil {
+			return fmt.Errorf("can't parse ip prefix \"%v\": %v", steering.L3Traffic.PrefixAddress, err)
+		}
+		if ip.To4() != nil { //IPv4 address
+			steerType = SteerTypeIPv4
+		}
+		tableID = steering.L3Traffic.FibTableID
+		prefixAddr = []byte(ip.To16())
+		ms, _ := ipnet.Mask.Size()
+		maskWidth = uint32(ms)
+	} else if steering.L2Traffic != nil {
+		steerType = SteerTypeL2
+		interfaceSwIndex, _, exists := swIfIndex.LookupIdx(steering.L2Traffic.InterfaceName)
+		if !exists {
+			return fmt.Errorf("for interface %v doesn't exist sw index", steering.L2Traffic.InterfaceName)
+		}
+		intIndex = interfaceSwIndex
+	}
+	req := &sr.SrSteeringAddDel{
+		IsDel:         boolToInt(delete),
+		TableID:       tableID,
+		BsidAddr:      bsidAddr,             //policy (to which we want to steer routing into) identified by binding sid (alternativelly it can be used SRPolicyIndex)
+		SrPolicyIndex: steering.PolicyIndex, //policy (to which we want to steer routing into) identified by policy index (alternativelly it can be used BsidAddr)
+		TrafficType:   steerType,            // type of traffic to steer
+		PrefixAddr:    prefixAddr,           // destination prefix address (L3 traffic type only)
+		MaskWidth:     maskWidth,            // destination ip prefix mask (L3 traffic type only)
+		SwIfIndex:     intIndex,             // incoming interface (L2 traffic type only)
+	}
+	reply := &sr.SrSteeringAddDelReply{}
+
+	if err := vppChan.SendRequest(req).ReceiveReply(reply); err != nil {
+		return err
+	}
+	if reply.Retval != 0 {
+		return fmt.Errorf("vpp call %q returned: %d", reply.GetMessageName(), reply.Retval)
+	}
+
+	log.WithFields(logging.Fields{"steer type": steerType, "L3 prefix address bytes": prefixAddr,
+		"L2 interface index": intIndex, "policy binding SID": bsidAddr, "policy index": steering.PolicyIndex}).
+		Debugf("%v steering to SR policy ", operationFinished)
+
+	return nil
+}
+
+func boolToInt(input bool) uint8 {
```

Comment:
This is actually boolTo**Uint** not *Int*.

Reply:
fixed in bd4f4f0a9c4c8cd2c5a2de517fa10850350067b2